assets/src/edit-story/app/media/local/test/useContextValueProvider.js

```diff
@@ -0,0 +1,263 @@
+        file: 'IMAGE-96x96.jpg',
+        width: 96,
+        height: 96,
+        mime_type: 'image/jpeg',
+        source_url:
+          'http://wp.local/wp-content/uploads/2020/09/IMAGE-96x96.jpg',
+      },
+      'web-stories-thumbnail': {
+        file: 'IMAGE-150x308.jpg',
+        width: 150,
+        height: 308,
+        mime_type: 'image/jpeg',
+        source_url:
+          'http://wp.local/wp-content/uploads/2020/09/IMAGE-150x308.jpg',
+      },
+      full: {
+        file: 'IMAGE.jpg',
+        width: 1080,
+        height: 2220,
+        mime_type: 'image/jpeg',
+        source_url: 'http://wp.local/wp-content/uploads/2020/09/IMAGE.jpg',
+      },
+    },
+    attribution: undefined,
+  },
+];
+
+describe('useContextValueProvider', () => {
+  let reducerState;
+  let reducerActions;
+  beforeEach(() => {
+    reducerState = {
+      processing: [],
+      processed: [],
+      media: [],
+      pageToken: '',
+      mediaType: '',
+      searchTerm: '',
+    };
+    reducerActions = {
+      fetchMediaStart: jest.fn(),
+      fetchMediaSuccess: jest.fn(),
+      fetchMediaError: jest.fn(),
+      resetFilters: jest.fn(),
+      setMedia: jest.fn(),
+      setMediaType: jest.fn(),
+      setSearchTerm: jest.fn(),
+      setNextPage: jest.fn(),
+      setProcessing: jest.fn(),
+      removeProcessing: jest.fn(),
+      updateMediaElement: jest.fn(),
+      deleteMediaElement: jest.fn(),
+    };
+
+    const useUploadMediaResult = {
+      uploadMedia: jest.fn(),
+      isUploading: jest.fn(),
+    };
+    useUploadMedia.mockImplementation(() => useUploadMediaResult);
+
+    const useUploadVideoFrameResult = {
+      uploadVideoFrame: jest.fn(),
+    };
+    useUploadVideoFrame.mockImplementation(() => useUploadVideoFrameResult);
+  });
+
+  it('resetWithFetch calls getMedia with cacheBust:true and then fetchMediaSuccess', async () => {
+    // Set up and make initial call to useContextValueProvider (which calls
+    // getMedia and fetchMediaSuccess once). Sets an empty media state.
+    const getMedia = jest.fn(() =>
+      Promise.resolve({
+        data: [],
+        headers: GET_MEDIA_RESPONSE_HEADER,
+      })
+    );
+    const apiState = {
+      actions: {
+        getMedia,
+      },
+    };
+    const configState = {
+      api: {},
+      allowedMimeTypes: {
+        video: [],
+      },
+    };
+    const { result } = renderHook(
+      () => useContextValueProvider(reducerState, reducerActions),
+      {
+        // eslint-disable-next-line react/display-name
+        wrapper: (params) => (
+          <ConfigProvider config={configState}>
+            <ApiContext.Provider value={apiState}>
+              {params.children}
+            </ApiContext.Provider>
+          </ConfigProvider>
+        ),
+      }
+    );
+
+    // This promise will only complete when the "done()" callback is called
+    // (see reducerActions.fetchMediaSuccess mock implementation in Promise).
+    await new Promise((done) => {
```

Comment:
Is this needed? It should be enough to 'await null' in the test for everything to finish, including the call to fetchMediaSuccess, something like:
```
getMedia.mockImplementation(() => ({
  data: GET_MEDIA_RESPONSE_BODY,
  headers: GET_MEDIA_RESPONSE_HEADER,
}));

result.current.actions.resetWithFetch();

await null;

expect(getMedia).toHaveBeenNthCalledWith(2, {
:
expect(reducerActions.fetchMediaSuccess).toHaveBeenNthCalledWith(2, {
:
```

Reply:
Done.